contracts/staking/TokenStaking.sol

```diff
@@ -0,0 +1,1427 @@
+        }
+    }
+
+    //
+    //
+    // Auxiliary functions
+    //
+    //
+
+    /// @notice Returns the authorized stake amount of the operator for the
+    ///         application.
+    function authorizedStake(address operator, address application)
+        external
+        view
+        override
+        returns (uint96)
+    {
+        return operators[operator].authorizations[application].authorized;
+    }
+
+    /// @notice Returns staked amount of T, Keep and Nu for the specified
+    ///         operator.
+    /// @dev    All values are in T denomination
+    function stakes(address _operator)
+        external
+        view
+        returns (
+            uint96 tStake,
+            uint96 keepInTStake,
+            uint96 nuInTStake
+        )
+    {
+        OperatorInfo storage operator = operators[_operator];
+        tStake = operator.tStake;
+        keepInTStake = operator.keepInTStake;
+        nuInTStake = operator.nuInTStake;
+    }
+
+    /// @notice Returns start staking timestamp for T stake.
+    /// @dev    This value is set at most once, and only when a stake is created
+    ///         with T tokens. If a stake is created from a legacy stake,
+    ///         this value will remain as zero
+    function getStartTStakingTimestamp(address operator)
+        external
+        view
+        returns (uint256)
+    {
+        return operators[operator].startTStakingTimestamp;
+    }
+
+    /// @notice Returns staked amount of NU for the specified operator
+    function stakedNu(address operator)
+        external
+        view
+        returns (uint256 nuAmount)
+    {
+        (nuAmount, ) = tToNu(operators[operator].nuInTStake);
+    }
+
+    /// @notice Gets the stake owner for the specified operator address.
+    /// @return Stake owner address.
+    function ownerOf(address operator) external view returns (address) {
+        return operators[operator].owner;
+    }
+
+    /// @notice Gets the beneficiary for the specified operator address.
+    /// @return Beneficiary address.
+    function beneficiaryOf(address operator)
+        external
+        view
+        returns (address payable)
+    {
+        return operators[operator].beneficiary;
+    }
+
+    /// @notice Gets the authorizer for the specified operator address.
+    /// @return Authorizer address.
+    function authorizerOf(address operator) external view returns (address) {
+        return operators[operator].authorizer;
+    }
+
+    /// @notice Checks if the specified operator has a stake delegated and if it
+    ///         has been authorized for at least one application. If this
+    ///         function returns true, off-chain client of the given operator is
+    ///         eligible to join the network.
+    function hasStakeDelegated(address operator)
+        external
+        view
+        override
+        returns (bool)
+    {
+        return operators[operator].authorizedApplications.length > 0;
```

Comment:
I think it will be part of each application. For example, in PRE we will check authorized amount each increase (and request decrease). Operator protected from Keep and NU legacy contract anyway.

Reply:
In this case maybe we should remove `hasStakeDelegated` altogether? We will save some space in the contract. Its docs are not really precise: 

> If this function returns true, off-chain client of the given operator is eligible to join the network.

And we assume this is application's responsibility.

Also, it's named `hasStakeDelegated` while we also check if there is at least one authorization for that delegated stake.

Reply:
Dracarys? 🔥

Reply:
I'm for removing 👍

Reply:
✔️